Tests/NIOHTTP2Tests/ConfiguringPipelineAsyncMultiplexerTests.swift

```diff
@@ -0,0 +1,167 @@
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import XCTest
+
+import NIOCore
+import NIOEmbedded
+import NIOHPACK
+import NIOHTTP1
+@_spi(AsyncChannel) import NIOHTTP2
+import NIOTLS
+
+final class ConfiguringPipelineAsyncMultiplexerTests: XCTestCase {
+    var clientChannel: NIOAsyncTestingChannel!
+    var serverChannel: NIOAsyncTestingChannel!
+
+    override func setUp() {
+        self.clientChannel = NIOAsyncTestingChannel()
+        self.serverChannel = NIOAsyncTestingChannel()
+    }
+
+    override func tearDown() {
+        self.clientChannel = nil
+        self.serverChannel = nil
+    }
+
+
+    static let requestFramePayload = HTTP2Frame.FramePayload.headers(.init(headers: HPACKHeaders([(":method", "GET"), (":authority", "localhost"), (":scheme", "https"), (":path", "/")]), endStream: true))
+    static let responseFramePayload = HTTP2Frame.FramePayload.headers(.init(headers: HPACKHeaders([(":status", "200")]), endStream: true))
+
+    final class OKResponder: ChannelInboundHandler {
+        typealias InboundIn = HTTP2Frame.FramePayload
+        func channelRead(context: ChannelHandlerContext, data: NIOAny) {
+            let frame = self.unwrapInboundIn(data)
+            switch frame {
+            case .headers:
+                break
+            default:
+                fatalError("unexpected frame type: \(frame)")
+            }
+
+            context.channel.writeAndFlush(responseFramePayload, promise: nil)
+
+            context.fireChannelRead(data)
+        }
+    }
+
+    final class SimpleRequest: ChannelInboundHandler, ChannelOutboundHandler {
+        typealias InboundIn = HTTP2Frame.FramePayload
+        typealias OutboundIn = HTTP2Frame.FramePayload
+
+        func writeRequest(context: ChannelHandlerContext) {
+            context.channel.writeAndFlush(requestFramePayload, promise: nil)
+        }
+
+        func channelActive(context: ChannelHandlerContext) {
+            self.writeRequest(context: context)
+            context.fireChannelActive()
+        }
+    }
+
+    func testBasicPipelineCommunicates() async throws {
+        let requestCount = 100
+
+
+        let (serverRecorderStream, serverRecorderContinuation) = AsyncStream<InboundRecorder<HTTP2Frame.FramePayload>>.makeStream()
+        var serverRecorderIterator = serverRecorderStream.makeAsyncIterator()
+
+        let clientMultiplexer = try await assertNoThrowWithValue(
+            try await self.clientChannel.configureAsyncHTTP2Pipeline(
+                mode: .client, connectionConfiguration: .init(), streamConfiguration: .init()) { channel -> EventLoopFuture<Channel> in
+                    self.serverChannel.eventLoop.makeSucceededFuture(channel)
+                }.get()
+        )
```

Comment:
Note also that this fixes the event loop used in the init:

```suggestion
            try await self.clientChannel.configureAsyncHTTP2Pipeline(
                mode: .client, 
                connectionConfiguration: .init(), 
                streamConfiguration: .init()
            ) { channel -> EventLoopFuture<Channel> in
                channel.eventLoop.makeSucceededFuture(channel)
            }.get()
        )
```